tasks/Task.cpp

```diff
@@ -47,22 +46,37 @@ void Task::onAnswer()
 
 void Task::onCandidate()
 {
-    string mid = mDecoder.getMid();
     string candidate = mDecoder.getCandidate();
-
-    try
+    if (mDecoder.validateMidFieldPresent())
```

Comment:
Having this huge block of code two times should make you uneasy (this is called a code smell)

Essentially, libdatachannel interprets an empty mid as "no mid". Just do

``` cpp
string mid;
if (mDecoder.isMidFieldPresent()) {
   mid = mDecoder.getMid();
}
try {
   mPeerConnection->addRemoteCandidate(rtc::Candidate(candidate, mid));
}
...
```

If it was not possible, you could have at least created a separate method that takes the candidate as argument, and call it with either Candidate objects.

Reply:
nice. Indeed, this kind of duplicated code block is not comfortable, but I didn't realize that we can pass the mid argument as empty.